server/src/session.ts

```diff
@@ -131,6 +131,51 @@ export class Session {
     // project as dirty to force update the graph.
     project.markAsDirty();
     this.info(`Enabling Ivy language service for ${project.projectName}.`);
+
+    // Send diagnostics since we skipped this step when opening the file
+    // (because language service was disabled while waiting for ngcc).
+    // Technically we could just call project.refreshDiagnostics(), but we need
+    // to handle the special case where all open files are external templates.
+    const openFiles = this.getOpenFilesForNewlyLoadedProject(project);
+    this.sendPendingDiagnostics(openFiles);
+  }
+
+  /**
+   * Get the open files for a newly loaded project. If all the open files are
+   * external templates, then try to match them to the specified `project` by
+   * including one of the root files. The inclusion of a root file triggers
+   * global analysis, during which we fix up the external templates.
+   */
+  private getOpenFilesForNewlyLoadedProject(project: ts.server.Project): string[] {
+    const openFiles: string[] = [];
+
+    const iterable = this.projectService.openFiles.keys();
+    for (let it = iterable.next(); !it.done; it = iterable.next()) {
+      const fileName = it.value;
+      const scriptInfo = this.projectService.getScriptInfo(fileName);
+      // External templates without a project are orphans.
+      if (scriptInfo?.isOrphan() || scriptInfo?.attachToProject(project)) {
+        openFiles.push(scriptInfo.fileName);
+      }
```

Comment:
At this point, shouldn't all external templates be orphans? What is the significance of `scriptInfo?.attachToProject(project)`?

Reply:
Line 157 here is wrong for two reasons (shouldn't have written code when I'm sleepy):
1. all the script infos are not actually orphans, because by default they get attached to an `inferred` project. 
    What we want to check here is really "if the script info is not attached to any configured project"
2. I meant to call `isAttachedTo(project)`, not `attachToProject()` 😅